src/Compilers/Core/CodeAnalysisTest/Collections/ImmutableSegmentedDictionaryBuilderTest.cs

```diff
@@ -84,21 +79,21 @@ public void BuilderFromMap()
         [Fact]
         public void SeveralChanges()
         {
-            var mutable = ImmutableDictionary<int, string>.Empty.ToBuilder();
+            var mutable = ImmutableSegmentedDictionary<int, string>.Empty.ToBuilder();
             var immutable1 = mutable.ToImmutable();
-            Assert.Same(immutable1, mutable.ToImmutable()); // "The Immutable property getter is creating new objects without any differences."
+            Assert.True(immutable1 == mutable.ToImmutable()); // "The Immutable property getter is creating new objects without any differences."
 
             mutable.Add(1, "a");
             var immutable2 = mutable.ToImmutable();
-            Assert.NotSame(immutable1, immutable2); // "Mutating the collection did not reset the Immutable property."
-            Assert.Same(immutable2, mutable.ToImmutable()); // "The Immutable property getter is creating new objects without any differences."
+            Assert.False(immutable1 == immutable2); // "Mutating the collection did not reset the Immutable property."
```

Comment:
Consider using `Equal`/`NotEqual` for these, rather than `True`/`False` with operators, as that will give better errors.

Reply:
This test is leveraging the fact that `==` implements reference equality for these collections (similar to `ImmutableArray<T>`). While `Equals` technically would do the same thing for this type, it feels one step further removed because it looks like value equality is intended by the test when we're really looking for reference equality.

Reply:
I don't believe that outweighs the benefits from getting clear error messages.

Reply:
⏱ will switch to `Assert.False(IsSame(...))` to match the other cases

Reply:
I don't believe that addresses my comments about error messages.

Reply:
This is correct, but also By Design in accordance with our adopted test framework (see https://github.com/xunit/xunit/issues/350 for additional discussion). The approach used by this pull request adheres to the test framework recommendation that the assertion call site clearly show the intent of the assertion, which `IsSame` will do.